### PR TITLE
Nomovok

### PR DESCRIPTION
**Pull request by timohannukkala**
Few changeses:
- Qt4 -> Qt5
- Sorting names can be based on lastname first/firstname first 
- Added addresses for user (work, home, other) - Add - View - Edit
- Added settings page - that allows to sort contacts -> all users -> that are favorite users -> that have phone number -> that have email address
- requires chances to libcontacts, qtcontacts-sqlite, nemo-qml-plugin-contacts, qmlcontacts
  (if you want to disable all these changes, except qt4 -> qt5, possible too, no need to pull library changes, just edit little qmlcontacts and fix error messages)

### CODE REVIEW

**Comment by rburchell** [comment]
woah. please read up on git rebase -i (and git commit --amend) to create somewhat of a sane history, please

File: rpm/qmlcontacts-qt5.spec [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#
```
Comment (rburchell) [inline rpm/qmlcontacts-qt5.spec:1]: don't fork the packaging for apps - just port it to qt5 (see the other ports for examples)

there's not enough people working on this code to justify having both qt4 and qt5 versions

File: src/main.cpp [hunk 2]
```diff
@@ -31,9 +31,9 @@
 
 #include <QDesktopServices>
 #include <QApplication>
```
Comment (rburchell) [inline src/main.cpp:33]: QGuiApplication, you want to drop QtWidgets too during the port (as it's a lot of dead weight)

File: src/main.cpp [hunk 3]
```diff
@@ -31,9 +31,9 @@
 
 #include <QDesktopServices>
```
Comment (rburchell) [inline src/main.cpp:32]: QStandardPaths

File: src/main.cpp [hunk 4]
```diff
@@ -45,15 +45,16 @@
...
 {
-    QApplication *application;
-    QDeclarativeView *view;
+    QQuickView *view;
 #ifdef HAS_BOOSTER
+    QGuiApplication *application;
     application = MDeclarativeCache::qApplication(argc, argv);
     view = MDeclarativeCache::qDeclarativeView();
 #else
+    QApplication *application;
```
Comment (rburchell) [inline src/main.cpp:54]: QGuiApplication

File: src/main.cpp [hunk 5]
```diff
@@ -45,15 +45,16 @@
...
     qWarning() << Q_FUNC_INFO << "Warning! Running without booster. This may be a bit slower.";
     QApplication stackApp(argc, argv);
```
Comment (rburchell) [inline src/main.cpp:56]: same

File: src/main.cpp [hunk 6]
```diff
@@ -77,15 +78,16 @@ int main(int argc, char **argv)
         view->setSource(QUrl::fromLocalFile("main.qml"));
     else
         view->setSource(QUrl("qrc:/qml/main.qml"));
-
+/*
```
Comment (rburchell) [inline src/main.cpp:81]: just delete this block, not needed anymore

File: src/qml/CheckBox.qml [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import QtQuick 2.0
```
Comment (rburchell) [inline src/qml/CheckBox.qml:1]: can you please split the CheckBox stuff/UI changes off, and let's just focus on the port for now?

File: src/qml/ContactCardPage.qml [hunk 8]
```diff
@@ -29,11 +29,10 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
  */
 
-import QtQuick 1.1
-import com.nokia.meego 1.0
+import QtQuick 2.0
+import com.nokia.meego 2.0
 import org.nemomobile.qmlcontacts 1.0
 import org.nemomobile.contacts 1.0
-import stage.rubyx.voicecall 1.0
```
Comment (rburchell) [inline src/qml/ContactCardPage.qml:36]: just removing this import will break the call button, won't it? you want import org.nemomobile.voicecall 1.0

File: src/qml/ContactListPage.qml [hunk 9]
```diff
@@ -52,6 +52,10 @@ Page {
              app.contactListModel.search(searchbox.searchText);
          }
      }
+
```
Comment (rburchell) [inline src/qml/ContactListPage.qml:55]: drop the empty function

File: src/qml/SettingsPage.qml [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
+ */
+
+import QtQuick 2.0
+import com.nokia.meego 2.0
+//import org.nemomobile.qmlcontacts 1.0
```
Comment (rburchell) [inline src/qml/SettingsPage.qml:34]: drop unused imports

File: src/qml/SettingsPage.qml [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+        text: qsTr("Settings")
+        color: "#007FFF"
+    }    
+    
+    Label {
+         id: searchbox
+         text: "Show contacts:"
+         anchors.top: header.bottom
+         width: parent.width
+         anchors.leftMargin: 20
```
Comment (rburchell) [inline src/qml/SettingsPage.qml:52]: UiConstants.DefaultMargin

File: src/qml/SettingsPage.qml [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+     }    
+
+     CheckBox {
+         id: showAllContacts
+         text: "Show All contacts"
+         anchors.top: searchbox.bottom
+         width: parent.width
+         anchors.leftMargin: 20
```
Comment (rburchell) [inline src/qml/SettingsPage.qml:60]: same: but isn't this technically breaking the width: parent.width?

File: src/qml/SettingsPage.qml [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+             app.contactListModel.setFilterType(1);
+         }
+    }
+     
+     CheckBox {
+         id: showFavoriteContacts
+         text: "Show favorite contacts"
+         anchors.top: showAllContacts.bottom
+         width: parent.width
+         anchors.leftMargin: 20
```
Comment (rburchell) [inline src/qml/SettingsPage.qml:76]: same for margin - I won't keep repeating this

File: src/qml/main.qml [hunk 14]
```diff
@@ -55,10 +55,12 @@ PageStackWindow {
 
     property PeopleModel contactListModel: PeopleModel {
 
-// for testing purposes
```
Comment (rburchell) [inline src/qml/main.qml:58]: revert this please

_edit_ oh, actually ok I see. can you just remove the importContacts line?

File: src/src.pro [hunk 15]
```diff
@@ -3,18 +3,18 @@ PROJECT_NAME = qmlcontacts
 TEMPLATE = app
 CONFIG += ordered mobility hide_symbols
 MOBILITY += contacts
-QT += declarative
+QT += quick widgets
```
Comment (rburchell) [inline src/src.pro:6]: drop qtwidgets

File: src/qml/main.qml [hunk 16]
```diff
@@ -55,10 +55,12 @@ PageStackWindow {
...
-//        Component.onCompleted: {
+    Component.onCompleted: {
 //            importContacts("../test/example.vcf")
-//        }
+
+              setDisplayLabelOrder(1);
```
Comment (rburchell) [inline src/qml/main.qml:61]: these settings are (or probably should) be controlled and persisted using nemo-qml-plugin-configuration's ConfigurationValue (which uses gconf as a backend at present)

File: rpm/qmlcontacts-qt5.spec (hunk 1 above)
Comment (timohannukkala) [inline rpm/qmlcontacts-qt5.spec:1]: Ok. when I read the history, I think, I did mess up with merging. Sorry, I fix those and try again.

**Comment by thp** [comment]
This has merge conflicts, and has not seen any updates from Robin's feedback. Closing this. Please reopen if you still want to get this in.

**Pull request closed without merging**